frontend/src/metabase/parameters/components/CardValuesSourceModal/CardStepModal.tsx

```diff
@@ -75,13 +80,40 @@ const CardStepModal = ({
       ]}
       onClose={onClose}
     >
-      <ModalBody>
-        <DataPickerContainer value={value} onChange={setValue} />
-      </ModalBody>
+      <DataPicker.Provider>
+        <DataPickerSearchInput />
+        <DataPickerContainer>
+          <DataPicker value={value} onChange={setValue} />
+        </DataPickerContainer>
+      </DataPicker.Provider>
     </ModalContent>
   );
 };
 
+const DataPickerSearchInput = () => {
+  const { search } = useDataPicker();
```

Comment:
Cool! Seems like the `DataPickerContainer` is already doing the heavy-lifting searching questions for us.